tests/test_parts.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020-2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+import pydantic
+import pytest
+
+from charmcraft import parts
+
+
+class TestPartValidation:
+    """Part data validation scenarios."""
+
+    def test_part_validation_happy(self):
+        data = {
+            "plugin": "make",
```

Comment:
`make` feels a bit odd as it's not owned by charmcraft. perhaps its going to be updated to charm plugin? :)

Reply:
Changed to test using the charm plugin.